add Model Registry doc to website (#3698)
kubeflow/website: content/en/docs/components/model-registry/overview.md

+++
title = "Overview"
description = "An overview for Kubeflow Model Registry"
weight = 10
                    
+++

{{% alpha-status
  feedbacklink="https://github.com/kubeflow/model-registry" %}}

## What is Model Registry ?

A model registry is an important component in the life cycle of AI/ML models, an integral component for any MLOps platform and for ML workflows.

A model registry provides a central index for ML model developers to index and manage models, versions, and ML artifacts metadata.
It fills a gap between model experimentation and production activities.
It provides a central interface for all stakeholders in the ML lifecycle to collaborate on ML models.

<img src="/docs/components/model-registry/images/index-mr-loop.png"
  alt="Model Registry MLOps loop"
  class="mt-3 mb-3">

DevOps, Data Scientists, and developers need to collaborate with other users in the ML workflow to get models into production.
Data scientists need an efficient way to share model versions, artifacts and metadata with other users that need to access to those models as part of the MLOps workflow.

<!--
## Architecture

tbd, take from Proposal
-->

## Next steps

> Follow the [getting-started guide](/docs/components/model-registry/getting-started/)

Reviewer: Can you fix the text?

Author: ![image](https://github.com/kubeflow/website/assets/1699252/3281d506-4f24-4b6d-a456-56fc50b1d436)

not sure I understood your request @rimolive ... which part need fixing? 🤔

Reviewer: Maybe `getting-started` should not have the `-` in the text, but it's just a nit thing. It's up to you.

+++
title = "Overview"
description = "An overview for Kubeflow Model Registry"
weight = 10
                    
+++

{{% alpha-status
  feedbacklink="https://github.com/kubeflow/model-registry" %}}

## What is Model Registry?

> A model registry is an important component in the life cycle of AI/ML models, an integral component for any MLOps platform and for ML workflows.

Reviewer 2: Do you want to re-use the ML lifecycle diagram that we are trying to add here: https://github.com/kubeflow/website/pull/3728 ?
E.g. we already re-used it in the Training Operator and Katib: https://www.kubeflow.org/docs/components/training/overview/#why-training-operator-

Author: To me, it sounds like this page can eventually accomodate more than 1 diagram... as can be seen in this PR thread, there are several concepts which might need expression in visual form:
- Model Registry being a central and pivotal component in the MLOps lifecycle, showcasing both inner and outer loop for MLOps
- Model Registry as "single pane of glass"
- Model Registry as involved in each task/phase of the MLOps workflow

So overall I believe we should eventually accomodate more than a single diagram, as in a single image sounds to me a bit difficult to be exactly like "everybody" want to see :) wdyt?
Should we progress with the agreement we can expand with more diagram as a next iteration?

Reviewer 2: I agree with you @tarilabs, I also want to modify this diagram to show that model registry plays role in Model Development, Model Tuning, Model Training, Model Serving stages.

Author: tracked in: https://github.com/kubeflow/model-registry/issues/109

marking as resolved this thread accordingly.

> A model registry is an important component in the life cycle of AI/ML models, an integral component for any MLOps platform and for ML workflows.

Reviewer 2: ```suggestion
A model registry is an important component in the lifecycle of AI/ML models, an integral component for any MLOps platform and for ML workflows.
```

## Use Cases

This section describes Model Registry use-cases in the context of a MLOps Platform with Model Training, Experimentation, and Deployment.

A company, ACME Inc., is developing a machine-learning model for predicting customer churn. They require a centralized model registry for their MLOps platform (based on Kubeflow) for managing their ML model development lifecycle, including training, experimentation, and deployment. They want to ensure model governance, reproducibility, and efficient collaboration across data scientists and engineers.

### Persona:

* **Data Scientist**: Responsible for developing and evaluating different models for customer churn prediction. They need to track the performance of various model versions and compare them easily.
* **MLOps Engineer**: Responsible for deploying the chosen model into production. They need to access the latest model version and its metadata to configure the deployment environment.
* **Business Analyst**: Monitors the deployed model's performance and makes decisions based on its predictions. They need to understand the lineage of the model and its operational metrics.

### Use Case 1: Tracking the Training of Models

> The_ Data Scientist_ uses Kubeflow Notebooks to perform exploratory research and trains several types of models, with different hyperparameters and metrics. The Kubeflow Model Registry is used to track those models, in order to make comparisons and identify the best-performing model. Once the champion model is selected, the _Data Scientist _shares the model with the team. The _Data Scientist_ also tracks the lineage of training data sources and notebook code.

Author: done with 0659ec5cd803685bf4054d79ec6f52146db62eb4

* Track models available on storage: once the model is stored, it can then be tracked in the Kubeflow Model Registry for managing its lifecycle. The Model Registry the registry can catalog, list, index, share, record, organize this information. This allows the _Data Scientist_ to compare different versions and revert to previous versions if needed.
* Track and compare performance: View key metrics like accuracy, recall, and precision for each model version. This helps identify the best-performing model for deployment.
* Create lineage: Capture the relationships between data, code, and models. This enables the _Data Scientist_ to understand the origin of each model and reproduce specific experiments.
* Collaborate: Share models and experiment details with the _MLOps Engineer_ for deployment preparation. This ensures a seamless transition from training to production.

### Use Case 2: Experimenting with Different Model Weights to Optimize Model Accuracy

> The _Data Scientist_ after identifying a base model, uses KubeFlow Pipelines, Katib, and other components to experiment model with alternative weights, hyperparameters, and other variations to improve the model’s performance metrics; Kubeflow Model Registry can be used to track data related to experiments and runs for comparison, reproducibility and collaboration.

Reviewer 3: ```suggestion
The _Data Scientist_ after identifying a base model, uses Kubeflow Pipelines, Katib, and other components to experiment model training with alternative weights, hyperparameters, and other variations to improve the model’s performance metrics. Kubeflow Model Registry can be used to track data related to experiments and runs for comparison, reproducibility and collaboration.
```

Author: done with 1ab5aa42b39a97b0efe705b6abdf147f764ac2e4

* Retrieve the latest model version: Easily access the model version approved for deployment.
* Access model metadata: Understand the model's architecture, hyperparameters, and performance metrics. This helps the MLOps engineer to configure the deployment environment and monitor performance after deployment.
* Manage serving configurations: Define how the model will be served to production applications and set up necessary resources.
* Track model deployments: Monitor the deployed model's performance and track its health over time. This allows the MLOps Engineer to identify potential issues and take corrective actions.

### Use Case 4: Monitoring and Governance

The Business Analyst uses Kubeflow Model Registry to audit deployed models, monitor model performance, track key metrics and identify when model is drifting or needs re-training; capabilities of model lineage enable identifying all related artifacts such as training which was used or the original training data.

> * View model performance metrics: Track key metrics in real-time to understand how the model is performing in production.

Reviewer 3: I this really something that the MR should do? This capability seems like part of an observability tool outside of the MR scope

Author: thanks for spotting, the idea is for Model Registry to be "single pane of glass" but not substituting to existing/appropriate tools. 

Implemented with 3ee7e42e7fc2edfa8c50ba390306212a4a3532e8

  alt="Model Registry MLOps loop"
  class="mt-3 mb-3">

- **Create**: during the creation phase, the Model Registry facilitates collaboration between different teams in order to track changes, experiment with different model architectures, and maintain a history of model iterations.

> - **Verify**: in the verification stage, the Model Registry can support in making sure that models undergo rigorous testing and validation before progressing further, maintaining a record of performance metrics and test results for each version.

Reviewer 3: "the Model Registry supports rigorous testing and validation"

### Persona:

* **Data Scientist**: Responsible for developing and evaluating different models for customer churn prediction. They need to track the performance of various model versions and compare them easily.
* **MLOps Engineer**: Responsible for deploying the chosen model into production. They need to access the latest model version and its metadata to configure the deployment environment.
* **Business Analyst**: Monitors the deployed model's performance and makes decisions based on its predictions. They need to understand the lineage of the model and its operational metrics.

### Use Case 1: Tracking the Training of Models

The_ Data Scientist_ uses Kubeflow Notebooks to perform exploratory research and trains several types of models, with different hyperparameters and metrics. The Kubeflow Model Registry is used to track those models, in order to make comparisons and identify the best-performing model. Once the champion model is selected, the _Data Scientist _shares the model with the team. The _Data Scientist_ also tracks the lineage of training data sources and notebook code.

> * Track models available on storage: once the model is stored, it can then be tracked in the Kubeflow Model Registry for managing its lifecycle. The Model Registry the registry can catalog, list, index, share, record, organize this information. This allows the _Data Scientist_ to compare different versions and revert to previous versions if needed.

Reviewer 4: This seems duplicated.

```suggestion
* Track models available on storage: once the model is stored, it can then be tracked in the Kubeflow Model Registry for managing its lifecycle. The Model Registry can catalog, list, index, share, record, organize this information. This allows the _Data Scientist_ to compare different versions and revert to previous versions if needed.
```

Author: oops thanks!

Author: oops thanks!
